### ISSUE TITLE
Does 'Disable Restrictions for Tab' button honour 'Untrusted' setting for domain?

### ISSUE THREAD

**Issue by jimbobmcgee** [hackademix/noscript#38]
I love the *Disable for Tab* feature, but recently got to wondering if, by using it, I am actually allowing third-party script domains that I've previously marked as *Untrusted* to run normally.

For example, if `google-analytics.com` is *Untrusted*, but *Disable for Tab* is clicked on a page that uses GA, is `google-analytics.com` now effectively *Temp Trusted*?

If so, would two states for *Disable for Tab* (i.e. including Untrusted and excluding Untrusted) be a good idea?  In particular where *not* trusting previously-Untrusted scripts was the default.

**Comment by hackademix** [comment]
Yes, I confirm disabling all restrictions means **all**, including those implied from the UNTRUSTED preset, and yes, an option like "Enforce the UNTRUSTED preset even when restrictions are disabled globally or for a tab." (even checked by default, to replicate NoScript 5's default behavior) is a good idea.
I'll try to implement it after 10.2, thanks.

**Comment by SchnWalter** [comment]
I've always felt the need for something on these lines, but this exact approach might cause a lot of confusion.

But, I'm thinking that besides the _Temp. Trusted_ and _TRUSTED_ buttons available **for each domain**, we could add a new _Disable for this tab_ button, something like this:

![image](https://user-images.githubusercontent.com/2186426/160287263-c5a60509-bd22-4070-a86d-7ba9218608f9.png)

For the domain in the screenshot, I need the first party script on several pages and on others I also need to enable a 3rd party script. But I want any change to go away as soon as I close the tab, I don't want the changes to linger on if I forget to revert them. Currently I can't do that, and I would very much like to avoid using the _Temp. Trusted_, _TRUSTED_ or the _Disable restrictions for this tab_ buttons.

If others find this useful, I could spend the time to implement it.

**Comment by jimbobmcgee** [comment]
@SchnWalter - if that is something you are still interested in/willing to implement, I could certainly see a use for it.  

I don't particularly like that, when _temp trusting_ something, it is automatically/retrospectively applied to all open and subsequent tabs, and being able to limit to just the current tab would certainly fix that.  (Ironically, I came today to ask for this very thing, and thought I'd check the progress of my old issues, only to find that you had suggested the very same thing I had come to ask!)

But for the purposes of _this_ issue, I would still like to be able to _disable restrictions_ for the current tab and choose whether it honours the existing _unstrusted_ setting or not. 

I'm not sure how far into the 10.3 milestone v11.4.26 _is_...:smile:

**Comment by ifrh** [comment]
Version 13.6.19 does not have the proposed icon 

<img width="1319" height="509" alt="Image" src="https://github.com/user-attachments/assets/c3908c15-2a0d-469b-a2fe-25224e3836d1" />
